exa_error.go

```diff
@@ -0,0 +1,75 @@
+package error_reporting_go
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+)
+
+type ErrorMessageBuilder struct {
+	errorCode  string
+	message    string
+	parameters [][]string
+	mitigation string
+}
+
+func ExaError(errorCode string) *ErrorMessageBuilder {
+	return &ErrorMessageBuilder{
+		errorCode: errorCode,
+	}
+}
+
+func (builder *ErrorMessageBuilder) Message(message string) *ErrorMessageBuilder {
+	builder.message = message
+	return builder
+}
+
+func (builder *ErrorMessageBuilder) Parameter(arguments ...interface{}) *ErrorMessageBuilder {
```

Comment:
Is there no other way express the optional parameter? I've trouble understanding this, since I don't know in which order to provide the arguments.

Reply:
interface{} disables static type checking. Could you use `...string` instead?

Reply:
How about another method `DescribedParameter`